pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -583,6 +590,30 @@ func (p *criStatsProvider) addSwapStats(
 	}
 }
 
+func (p *criStatsProvider) addIOStats(
+	ps *statsapi.PodStats,
+	podUID types.UID,
+	allInfos map[string]cadvisorapiv2.ContainerInfo,
+	cs *statsapi.ContainerStats,
+) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.KubeletPSI) {
+		return
+	}
+	// try get IO stats from cadvisor first.
+	podCgroupInfo := getCadvisorPodInfoFromPodUID(podUID, allInfos)
+	if podCgroupInfo != nil {
+		ps.IO = cadvisorInfoToIOStats(podCgroupInfo)
+		return
+	}
+
+	if cs.IO != nil {
+		if ps.IO == nil {
+			ps.IO = &statsapi.IOStats{Time: cs.IO.Time}
+		}
+		// Pod level PSI stats cannot be calculated from container level
+	}
```

Comment:
I think these should be reversed

Reply:
Could you elaborate why? I was following the suit from other methods like `addPodCPUMemoryStats` and `addSwapStats`

Reply:
ah okay that's fine for now then